libraries/botbuilder-core/src/turnContext.ts

```diff
@@ -707,6 +708,20 @@ export class TurnContext {
         return this._respondedRef.responded;
     }
 
+    /**
+     * Gets the locale stored in the turnState.
+     */
+    public get locale(): string {
+        return this._turnState.get(this._turnLocale);
```

Comment:
I"m not sure we want to use turnState for locale like this.  Please explain why this is necessary.

Reply:
This is the same way we did in C#.

Reply:
Resolving via https://github.com/microsoft/botbuilder-js/pull/2692/files#r469650213